Add Harkonnen 6b

MustaphaTR opened the pull request.

MustaphaTR committed:
Add Harkonnen 6b

ltem commented on mods/d2k/maps/harkonnen-06b/harkonnen06b.lua:
```diff
@@ -0,0 +1,200 @@
+	{ OrdosEntry9.Location, OrdosRally9.Location }
+}
+
+OrdosInterval =
```
Maybe I overlook something but what is `OrdosInterval` used for?

MustaphaTR replied:
`FremenInterval` was used by Sietchs producing fremen on All4 and Har4, it is not used in this, or other maps, looks like i just renamed and forget about it. I think 6a is merged with that, i'll remove it from both.

ltem reviewed (commented):
The mission plays like the original. Only the `Smugglers` will become your enemies if you attack their units (message pops up (see image)) but they won't send units against you. I've overlooked this in `Harkonnen06a` but it doesn't really change the gameplay ~and I think this isn't possible in OpenRA for now~. 
Edit1: I don't know how but learned it is possible to do in Lua, so maybe this can be added (later).
 
|![harkonnen6b_smugglershostile](https://user-images.githubusercontent.com/20945684/32691949-0b2e496c-c710-11e7-8ac5-86e47a87554f.png)|
|:---:|

After this is I'll approve this PR.

MustaphaTR commented:
I intentedly left out Smuggler side changing, in 6a too. It is not really important and lack of alliance changes make it unnecessarily complicated.

RA and TD mods campaign missions use Enemy Unit/Structure and no owner name. I'm planning to add that to D2K too, probably when Har Missions are all added. Maybe i add Smuggler Owner changes with it too.

pchote commented:
Smugglers can be made hostile by setting up a second faction with different allies/enemies and switching all the actors over using lua.

ltem commented:
Okay, I'm fine with that.

abcdefg30 commented:
> Smugglers can be made hostile by setting up a second faction with different allies/enemies and switching all the actors over using lua.

Imho that is what should be done to retain the " `Smugglers` will become your enemies if you attack their units".

MustaphaTR commented:
Updated, added the Smuggler Owner change to both 6th mission, but still a bit different than original. You can't normally target the Smuggler units, but if you force fire on them or hit them by mistake, they turn against you.

Also Ordos may mistakely hit them and they can be enemy to ordos, without being an enemy to you. I remember that being case in the original, but not 100% sure, but it makes more sense this way.

ltem reviewed (approved):
Requests fixed and no more issues found.

reaperrr commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b44c03487b733e61d79bd3ce1fea1a81dc24646c...618005733452cf299625ba40665917ab6b7895b5)
